src/rust/cryptography-x509-validation/src/policy.rs

```diff
@@ -0,0 +1,66 @@
+// This file is dual licensed under the terms of the Apache License, Version
+// 2.0, and the BSD License. See the LICENSE file in the root of this repository
+// for complete details.
+
+use std::time::SystemTime;
+
+use cryptography_x509::certificate::Certificate;
+
+use crate::trust_store::{Store, StoreError};
+
+// /// A `Subpolicy` describes specific sets of explicitly permitted
+// /// and forbidden policy components.
+// ///
+// /// This name is bad.
+// struct Subpolicy<T> {
+//     permits: Vec<T>,
+//     forbids: Vec<T>,
+// }
+
+/// A `Policy` describes user-configurable aspects of X.509 path validation.
+pub struct Policy<'a> {
+    time: SystemTime,
+
+    // TODO(ww): This is slightly wrong; `Store` allows root CAs while
+    // the untrusted intermediate set should enforce only intermediates
+    // as an invariant.
+    untrusted_intermediates: Store<'a>,
```

Comment:
Flagging for review: this is the wrong type, just here as a placeholder to let me stub more things out. I'm going to think some more about how best to (if at all) reuse the `Store` type for non-trusted contexts like this one.